Sources/TuistDependencies/Carthage/Interactors/CarthageFrameworksInteractor.swift

```diff
@@ -0,0 +1,103 @@
+import Foundation
+import TSCBasic
+import TuistCore
+import TuistSupport
+
+/// Saving frameworks installed using `carthage`.
+public protocol CarthageFrameworksInteracting {
+    /// Saves frameworks installed using `carthage`.
+    /// - Parameters:
+    ///   - carthageBuildDirectory: The path to the directory that contains the `Carthage/Build/` directory.
+    ///   - dependenciesDirectory: The path to the directory that contains the `Tuist/Dependencies/` directory.
+    func copyFrameworks(carthageBuildDirectory: AbsolutePath, dependenciesDirectory: AbsolutePath) throws
+}
+
+public final class CarthageFrameworksInteractor: CarthageFrameworksInteracting {
+    private let fileHandler: FileHandling
+
+    public init(fileHandler: FileHandling = FileHandler.shared) {
+        self.fileHandler = fileHandler
+    }
+
+    public func copyFrameworks(carthageBuildDirectory: AbsolutePath, dependenciesDirectory: AbsolutePath) throws {
```

Comment:
As mentioned in another, I wouldn't know any graph-related work in this PR. The graph generation should be something generic enough to work with other dependency managers.